src/actions/attack.cpp

```diff
@@ -1636,7 +1636,7 @@ int generic_combat_modifier(int lawful_bonus, unit_alignments::type alignment, b
 bool backstab_check(const map_location& attacker_loc,
 		const map_location& defender_loc,
 		const unit_map& units,
-		const std::vector<team>& teams)
+		const team_list& teams)
```

Comment:
Code later in this function will need an update, but the main reason for commenting on a specific line is to make a thread in the Github UI.

I feel a first refactor should be to ask what the code using the teams needs to know, and make an encapsulation of that which doesn't require the callers to know about any off-by-one convention. For example, backstab_check only needs an object that it can call this method on:
```
/**
 * Returns true if the two units belong to opposing sides, false if they're on allied sides or the same side.
 * If either unit has an invalid side number, it's an enemy of any unit except those with the identical invalid number.
 */
bool is_enemy(const unit& a, const unit& b) const;
```